### DISCUSSION TITLE
How to produce a usable iso out of this repository?

### DISCUSSION THREAD

**Discussion by Guelakais** [Jovian-Experiments/Jovian-NixOS#422] [Q&A] [answered]
I've tried to run:
```sh
nix-build -A isoGnome
```
which produced:
```sh
error: attribute 'isoGnome' in selection path 'isoGnome' not found
```

**Comment by samueldr** [comment]
You *generally wouldn't*... but I'm not saying you shouldn't.

To help us help you:

 - can you explain the reasoning that led you to want to produce an iso?
 - can you explain how you got to try `nix-build -A isoGnome`?
 - can you detail the overall goal you are trying to achieve here?

**Comment by Guelakais** [comment]
* I thought the whole idea of [Jovian-NIXOS](https://github.com/Jovian-Experiments/Jovian-NixOS) is to install another operating system on the Steam deck. What else is it for? I'm a bit confused to be honest.
* I got it from [here](https://heywoodlh.io/nixos-steamdeck) 
* I am trying to generate a bootable iso image, which I then want to put on a USB stick. I want to have jovian os on my steam deck.

**Reply by samueldr** [reply]
Thanks, the questions might have seemed a bit abrupt, but they greatly help me help you.

* * *

Right, I understand the steps you have taken. Though the guide you followed is from *January 3, 2023*, a lot has changed since.

Jovian NixOS is a configuration that is meant to be composed on top of a NixOS system. As such, you would install NixOS *as you would usually on another system*, on the Steam Deck first. And then, the configuration can be imported in your system configuration to get the Steam OS experience.

See the [getting started page](https://jovian-experiments.github.io/Jovian-NixOS/getting-started.html).

At this point in time, using Jovian NixOS presupposes some level of experience with NixOS.

**Comment by Guelakais** [comment, marked as answer]
`sudo nixos-rebuild switch` does work now

**Comment by ghost** [comment]
I do it this way:

I have a [configuration](https://codeberg.org/BroomBear/Vontoo_Install) specifically for ISOs.

You can generate one for either a Vanilla desktop or a SteamDeck. You do this by using this command: `nix build .#nixosConfigurations.{Vanilla or SteamDeck}.config.system.build.isoImage`

This ISO has a builtin shell script, launched by entering `start` in the shell, which installs the [actual system](https://codeberg.org/BroomBear/Vontoo), on the disk you point the setup to.

This is mainly meant for me and acquaintances, however it is very modular.
